CONTRIBUTING.md

```diff
@@ -21,6 +21,14 @@ or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any addi
 ## Recommended Tools
 * Visual Studio and Visual Studio Code https://visualstudio.microsoft.com/
 
+## Git
+
+### Conventions
+TODO
```

Comment:
We have not adopted a branch naming schema, but I think it would be nice to do so - it makes seeing all the branches locally a lot easier if they are sorted by user.

Reply:
At the scale of some internal Microsoft repos that seems really helpful, but for the size of this repo it seems like an unnecessary extra prefix to me. The lexicographic sorting is handy, but not enough of a feature for me to want to type the extra prefix when I create a branch. It communicates ownership, but not purpose. If ownership is important there's always forks. Just my $0.02. 🤷‍♂️

Reply:
To me the prefixed names, even at our scale, make reading this easier:
```
❯ git br -a
* main
  remotes/origin/HEAD -> origin/main
  remotes/origin/Send_AuthFlowResults
  remotes/origin/add-iwa
  remotes/origin/add_authflow_name
  remotes/origin/example-custom-events
  remotes/origin/main
  remotes/origin/new_telemetry_schema
  remotes/origin/simple-time
  remotes/origin/telemetry_schema_changes_1
  remotes/origin/user/shakhare/refactorPart1
  remotes/origin/users/haashah/contributing_git_conventions
  remotes/origin/users/ruitang/sigterm
  remotes/origin/users/ruitang/upgrade_dotnet
  remotes/origin/users/ruitang/upgrade_msal_4.44
```
* given tooling easily solves the "I have to type more issue" it generally shouldn't be a burden. (i.e. loki)
* Given it is required in many Msft repos, it's easier IMO to just stay consistent instead of changing across repos. I don't feel so strongly as to say we should require this and I would be fine documenting that "Some of us choose to be consistent with user/name/ to be consistent across other internal repos, but it is not required here"